hazelcast/src/main/java/com/hazelcast/instance/NodeState.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.instance;
+
+/**
+ * Possible states of a {@link Node} during its lifecycle.
+ * Some actions/operations may be allowed or denied
+ * according to current state of the {@link Node}.
+ *
+ * @see Node#start()
+ * @see Node#shutdown(boolean)
+ * @see com.hazelcast.spi.impl.AllowedDuringShutdown
+ *
+ * @since 3.6
+ */
+public enum NodeState {
+
+    /**
+     * Initial state of the Node. An {@code ACTIVE} node is allowed
```

Comment:
no change here. Currently node activated during start process. It's same in this branch too.